File: 152.java
class Problem152 {
    public int maxProduct(int[] nums) {
        int MAO = nums[0], mio = nums[0], mao = nums[0];
        for (int i = 1; i < nums.length; i++){
            if (nums[i] < 0){
                int temp = mio;
                mio = mao;
                mao = temp;
            }
            mio = Math.min(nums[i], mio*nums[i]);
            mao = Math.max(nums[i], mao*nums[i]);
            MAO = Math.max(MAO, mao);
        }
        return MAO;
    }
}

File: 152.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Problem152 {
public:
    int maxProduct(vector<int>& nums) {
        int m = INT_MIN;
        for (int i = 0; i < nums.size(); i++){
            int curr = nums[i];
            m = max(m, curr);
            for (int j = i+1; j < nums.size(); j++){
                curr *= nums[j];
                m = max(m, curr);
            }
        }
        return m;
    }
};